diff --git a/fs/dcache.c b/fs/dcache.c
index e795821a328f..0f045cb40102 100644
--- a/fs/dcache.c
+++ b/fs/dcache.c
@@ -2217,6 +2217,11 @@ struct dentry *__d_lookup_rcu(const struct dentry *parent,
 			continue;
 		if (dentry_cmp(dentry, str, hashlen_len(hashlen)) != 0)
 			continue;
+#ifdef CONFIG_KSU_SUSFS_SUS_PATH
+			if (dentry->d_inode && unlikely(dentry->d_inode->i_state & 16777216) && likely(current_cred()->user->android_kabi_reserved1 & 16777216)) {
+				continue;
+			}
+#endif
 		*seqp = seq;
 		return dentry;
 	}
